src/client/opamAdminCommand.ml

```diff
@@ -1005,13 +1005,38 @@ let add_constraint_command =
         let deps =
           OpamFormula.map (function
               | (n,c as atom) ->
-                if n = name then Atom (n, (add_cstr nv n c))
+                if n = name then Atom (n, (add_cstr cstr nv n c))
                 else Atom atom)
             deps0
         in
-        if deps <> deps0 then
+        let depopts0 = OpamFile.OPAM.depopts opam in
+        let conflicts0 = OpamFile.OPAM.conflicts opam in
+        let contains =
+          OpamFormula.fold_left (fun contains (n,_) ->
+              contains || n = name) false
+        in
+        let conflicts =
+          if contains depopts0 then
+            let icstr =
+              match cstr with
+              | OpamFormula.Atom (Constraint (relop, FString v)) ->
+                OpamFormula.Atom (Constraint (OpamFormula.neg_relop relop, FString v))
+              | _ -> OpamFormula.Empty
```

Comment:
Shouldn't this be an `assert false` ?
That or just use `OpamFormula.neg (fun (op,v) -> OpamFormula.neg_relop op, v) cstr` ?
(See `admin-scripts/depopts_to_conflicts.ml`)

Reply:
better use the first `cstr` option definition to have only the two ways: a constraint is done → construct its neg, a constraint is released → remove package from conflicts (this case wasn't handled)